components/function-controller/cmd/registry-gc/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/docker/distribution/reference"
+	"github.com/kyma-project/kyma/components/function-controller/internal/registry"
+	"github.com/vrischmann/envconfig"
+	ctrl "sigs.k8s.io/controller-runtime"
+	ctrlzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type config struct {
+	Namespace               string `envconfig:"default=kyma-system"`
+	RegistryConfigSecreName string `envconfig:"default=serverless-registry-config-default"`
+}
+
+func main() {
```

Comment:
IMO this function is hard to read because is too long. It has at least two functionalities: collecting image names from deployments, removing images with known name and unknown tag. What do you think about extracting these parts to separate functions?